de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/shipmentschedule/spi/impl/InOutProducerFromShipmentScheduleWithHU.java

```diff
@@ -414,10 +417,77 @@ public void setTrxItemProcessorCtx(final ITrxItemProcessorContext processorCtx)
 	@Override
 	public void process(final IShipmentScheduleWithHU item) throws Exception
 	{
+		updateShipmentDate(currentShipment, item);
 		createUpdateShipmentLine(item);
 		lastItem = item;
 	}
 
+	private void updateShipmentDate(@NonNull final I_M_InOut shipment, @NonNull final IShipmentScheduleWithHU candidate)
+	{
+		final I_M_ShipmentSchedule schedule = candidate.getM_ShipmentSchedule();
+		final Timestamp candidateShipmentDate = calculateShipmentDate(schedule, shipmentDateToday);
+
+		// the shipment was created before but wasn't yet completed;
+		if (isCandidateShipmentDateBestForShipment(shipment, candidateShipmentDate))
+		{
+			shipment.setMovementDate(candidateShipmentDate);
+			shipment.setDateAcct(candidateShipmentDate);
+
+			InterfaceWrapperHelper.save(shipment);
+		}
+	}
+
+	@VisibleForTesting
+	static boolean isCandidateShipmentDateBestForShipment(final @NonNull I_M_InOut shipment, final @NonNull Timestamp candidateShipmentDate)
```

Comment:
i'm not sure that the the `isCandidateShipmentDateBestForShipment` is about..i also skipped down to the tests and still don't really get it..
also it took me some time to even see what currentShipmentDate and candidateShipmentDate stand for...mabe rename currentShipmentDate  to movementDate..and extract those if-expressions into local variables..maybe then I see more easily what make a particular date the best..

Reply:
Done the variable part and also neramed the method with isShipmentDeliveryDateBetterThanMovementDate